adapter/internal/proxyserver/handler.go

```diff
@@ -0,0 +1,217 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+	Package handler contains logic related to authenticate the GA API requests and handle API requests accordingly.
+*/
+
+package proxyserver
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/wso2/product-microgateway/adapter/config"
+	logger "github.com/wso2/product-microgateway/adapter/internal/loggers"
+)
+
+// HTTPGetToGAHandler function handles get requests
+func (s *Server) HTTPGetToGAHandler(w http.ResponseWriter, r *http.Request) {
+	buildRequestToGA(s.client, w, r, http.MethodGet, nil)
+}
+
+// HTTPPostToGAHandler function handles post requests
+func (s *Server) HTTPPostToGAHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
We only need a single function right? let's remove everything else.

Reply:
Ack.
I will keep only the GET resource.